crates/matrix-sdk/src/test_utils/mocks/mod.rs

```diff
@@ -3243,6 +3243,244 @@ impl<'a> MockEndpoint<'a, LoginEndpoint> {
     }
 }
 
+/// A struct for building a login response with status code 200
+#[derive(Default)]
+pub struct LoginResponseBuilder200 {
+    access_token: Option<String>, // required
+    device_id: Option<String>,   // required
+    expires_in_ms: Option<u32>,
+    refresh_token: Option<String>,
+    user_id: Option<String>,    // required
+    homeserver_url: Option<String>, // required if well_known used
+    identity_url: Option<String>, // required if well_known and m.identity_server used
+}
+
+impl LoginResponseBuilder200 {
+    /// constructor for empty reponse 
+    pub fn new() -> Self { Self::default() }
+
+    /// sets access_token
+    pub fn set_access_token(&mut self, value: &str) -> &mut Self { 
```

Comment:
There's two ways to create a builder pattern, one mutates `self` the other takes ownership of `self` and returns `self`.

We already have a similar template here, so let's be consistent and use the later:

https://github.com/matrix-org/matrix-rust-sdk/blob/72d133260c29e132a75597ae6ca77963813e74fa/crates/matrix-sdk/src/test_utils/mocks/mod.rs#L3136-L3160